lumicks/pylake/kymo.py

```diff
@@ -380,6 +408,35 @@ def pixelcount_factory(_):
         result._pixelcount_factory = pixelcount_factory
         return result
 
+    def calibrate_to_kbp(self, length_kbp):
+        """Calibrate from microns to other units.
+
+        Parameters
+        ----------
+        length : int
+            length of the kymo in kilobase pairs
```

Comment:
Is this the most convenient way of specifying the calibration factor?

It seems convenient when cropping. Inconvenient when not cropping (because then you have to extrapolate from the bead to bead region to the whole kymo, since the kbps reflect the distance between the beads).

Also, `int`? Shouldn't kbp's be `float` or `bp` (`int`)?

Reply:
Do you have something else in mind that would make more sense?

For the current implementation, to me, it really only makes sense to calibrate if you crop out the beads. Otherwise you would have to set the origin and end point at the bead edges since the units don't make sense in these regions anyway. This was the reason I made the small `PositionCalibration` class, which I envisioned would be for "simple" calibrations like this.

For more involved calibrations (such as using landmarks) we could have another class to handle that but that exposes the same API to be used internally by `Kymo`

Reply:
I find it surprising that a variable called `length` should _not_ be `int`.

Reply:
CS-wise ok sure because you're usually talking about length of an array. But for a physical measurement, length is almost always a `float`. The only reason it was `int` for base pairs is because you can't have a fraction of a nucleotide

Reply:
@rpauszek Fair enough. That makes sense.
@mikhas So are you 1 or 2 meters tall? :D